extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -0,0 +1,156 @@
+import * as sinon from 'sinon';
+import { expect } from 'chai';
+import { CancellationToken, extensions } from 'vscode';
+import { CodeQLExtensionInterface } from '../../../extension';
+import { logger } from '../../../logging';
+import * as config from '../../../config';
+import * as ghApiClient from '../../../remote-queries/gh-api/gh-api-client';
+import { Credentials } from '../../../authentication';
+import * as fs from 'fs-extra';
+
+import { VariantAnalysisManager } from '../../../remote-queries/variant-analysis-manager';
+import {
+  VariantAnalysis as VariantAnalysisApiResponse,
+  VariantAnalysisScannedRepository as ApiVariantAnalysisScannedRepository
+} from '../../../remote-queries/gh-api/variant-analysis';
+import { createMockApiResponse } from '../../factories/remote-queries/gh-api/variant-analysis-api-response';
+import { createMockScannedRepos } from '../../factories/remote-queries/gh-api/scanned-repositories';
+import { createMockVariantAnalysisRepoTask } from '../../factories/remote-queries/gh-api/variant-analysis-repo-task';
+
+describe('Variant Analysis Manager', async function() {
+  let sandbox: sinon.SinonSandbox;
+  let cancellationToken: CancellationToken;
+  let variantAnalysisManager: VariantAnalysisManager;
+  let variantAnalysis: VariantAnalysisApiResponse;
+  let scannedRepos: ApiVariantAnalysisScannedRepository[];
+  let getVariantAnalysisRepoStub: sinon.SinonStub;
+  let getVariantAnalysisRepoResultStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    sandbox = sinon.createSandbox();
+    sandbox.stub(logger, 'log');
+    sandbox.stub(config, 'isVariantAnalysisLiveResultsEnabled').returns(false);
+    sandbox.stub(fs, 'mkdirSync');
+    sandbox.stub(fs, 'writeFile');
+
+    cancellationToken = {
+      isCancellationRequested: false
+    } as unknown as CancellationToken;
```

Comment:
Since this is running in VSCode, could we use a real cancellation token instead using the [`CancellationTokenSource`](https://code.visualstudio.com/api/references/vscode-api#CancellationTokenSource)?

Reply:
Sure! It does seem a bit shady to use `unknown`. Have added the change [here](https://github.com/github/vscode-codeql/pull/1559/commits/b82cd8b6f4a5875ba46e2165e965c37a83bfa50e).

Reply:
Thanks! Does it make sense to not mock the cancellation token at all? Instead of mocking the cancellation token source, can we just call `new CancellationTokenSource()`?

Reply:
Ah I see what you mean! I'll take a look at this in a separate issue.